src/main/java/frc/robot/commands/TurnToHeading.java

```diff
@@ -0,0 +1,71 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.commands;
+
+import java.util.Optional;
+import java.util.function.DoubleSupplier;
+
+import com.ctre.phoenix6.mechanisms.swerve.utility.PhoenixPIDController;
+
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj2.command.Command;
+
+import frc.robot.subsystems.DriveTrain;
+
+public class TurnToHeading extends Command {
+
+  private final DriveTrain m_driveTrain;
+  private ChassisSpeeds chassisSpeeds;
+  private PhoenixPIDController turnHeadingPID;
+  private double m_wantedHeading;
+  private final DoubleSupplier m_translationXSupplier;
+  private final DoubleSupplier m_translationYSupplier;
+
+  /**
+   * Creates a new autoAim.
+   */
+  public TurnToHeading(DriveTrain driveTrain, double wantedHeading, DoubleSupplier translationXSupplier,
+      DoubleSupplier translationYSupplier) {
+    m_driveTrain = driveTrain;
+    m_wantedHeading = wantedHeading;
+    this.m_translationXSupplier = translationXSupplier;
+    this.m_translationYSupplier = translationYSupplier;
+
+    turnHeadingPID = new PhoenixPIDController(0.1, 0, 0);// random number TODO
```

Comment:
You can't use a Phoenix PID. That is for on a Falcon motor. You want a WPILib based PID.

Reply:
Also, the constants should be defined separately.